File: 11classjava/src/main/java/com/ylee/a11classjava/Car.java
package com.ylee.a11classjava;

public  class Car extends Vehicle {
    private final String engineType;

    public Car(String name, int speed, int maxspeed,
               int maxFuelTank, int numberOFWheels,
               boolean hasABS, String engineType) {
        super(name, speed, maxspeed, maxFuelTank,
                numberOFWheels, hasABS);
        this.engineType = engineType;
    }

    @Override
    public String sound() {
        return "Sound from Car";
    }

    @Override
    public String soundabstract() {
        return "abstract sound from Car";
    }
}

File: 11classkotlin/src/main/java/com/ylee/a11classkotlin/Car.kt
package com.ylee.a11classkotlin

class Car : Vehicle {
    var engineType : String

    constructor(name: String, speed: Int, maxspeed: Int,
                maxFuelTank: Int, numberOfWheels: Int,
                hasABS: Boolean, engineType: String) :
            super(name, speed, maxspeed, maxFuelTank,
                    numberOfWheels, hasABS) {
        this.engineType = engineType
    }

    override fun sound(): String {
        return "Sound from Car"
    }

    override fun soundabstract() {

    }
}